src/quiz.js

```diff
@@ -2,75 +2,89 @@ import QuizUI from "./ui/QuizUI";
 import WordService from "./service/WordService";
 import Util from "./lib/Util";
 import Progress from "./model/Progress";
-import {LOCALSTORAGE_KEY} from './Constant';
+import { CURR_USER_KEY } from "./Constant";
+import EasyHttp from "./lib/EasyHttp";
+import User from "./model/User";
 
 const quizUi = new QuizUI(document);
 const wordService = new WordService();
-let correctWord;
-let wordCount = 0;
+const userHttp = new EasyHttp("http://localhost:3000/users");
+const userId = localStorage.getItem(CURR_USER_KEY);
 
-async function* wordGenerator() {
-  while (true) {
-    const wordArray = await wordService.getTodayWordByUserId(JSON.parse(localStorage.getItem(LOCALSTORAGE_KEY)));
-    if (wordArray.length === 0) {
-      return;
-    }
-    for (let word of wordArray) {
-      yield word;
+userHttp.getById(userId).then((plainObj) => {
+  const user = User.parseJSON(plainObj);
+
+  let correctWord;
+  let wordCount = 0;
+
+  async function* wordGenerator() {
+    while (true) {
+      const wordArray = await wordService.getTodayWordByUserId(JSON.parse(localStorage.getItem(CURR_USER_KEY)));
+      if (wordArray.length === 0) {
+        return;
+      }
+      for (let word of wordArray) {
+        yield word;
+      }
     }
   }
-}
 
-const generator = wordGenerator();
+  const generator = wordGenerator();
+
+  async function showWord() {
+    const wordObj = await generator.next();
+
+    const word = wordObj.value;
+    const done = wordObj.done;
 
-async function showWord() {
-  const wordObj = await generator.next();
+    if (done) {
+      if (wordCount === 0) {
+        quizUi.noWordLeft();
+        return;
+      }
 
-  const word = wordObj.value;
-  const done = wordObj.done;
+      quizUi.finish();
 
-  if (done) {
-    if (wordCount === 0) {
-      quizUi.noWordLeft();
+      const date = Util.getTodayDate();
+      const dayApart = user.findDayApart(date);
+      user.updateStreak(dayApart, date);
+      userHttp.update(userId, user);
```

Comment:
Is updating the streak necessarily part of showing a word?

Reply:
I put that there because when you finish the quiz you want to update streak.